### PR TITLE
Enable inference func to bert(finetune_classifier)

### PR DESCRIPTION
**Pull request by pengxin99**
## Description ##
Enable inference for finetune_classifier on the MRPC dataset. Add other module for inference:

- modle.save_parameters (save training parameters for inference)
- mx.profiler (do profiler for inference)

add two argument:

- save_dir (directory path to save the final model.)
- run_type (train or inference, default is train.)

before doing inference, please first doing training at least once to save parameters,and then use this code to run inference:
`MXNET_GPU_MEM_POOL_TYPE=Round GLUE_DIR=glue_data python finetune_classifier.py --task_name MRPC --batch_size 32 --gpu --run_type inference`

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: scripts/bert/finetune_classifier.py [hunk 1]
```diff
@@ -346,13 +362,64 @@ def train(metric):
...
                     step_loss / args.log_interval, \
-                    trainer.learning_rate, *metric_val))
+                    trainer.learning_rate, *metric_val,))
                 step_loss = 0
+
         mx.nd.waitall()
         evaluate(metric)
         toc = time.time()
         logging.info('Time cost={:.1f}s'.format(toc - tic))
-        tic = toc
```
Comment (TaoLv) [inline scripts/bert/finetune_classifier.py:355]: why remove this?

File: scripts/bert/finetune_classifier.py [hunk 2]
```diff
@@ -346,13 +362,64 @@ def train(metric):
...
+                    ','.join([i + ':{:.4f}' for i in metric_nm])
+                logging.info(eval_str.format(epoch_id + 1, batch_id + 1, len(train_data), \
+                    step_loss / args.log_interval, \
+                    *metric_val,))
+                step_loss = 0
+
+        mx.nd.waitall()
+
+    toc = time.time()
+    logging.info('Time cost={:.1f}s'.format(toc - tic))
```
Comment (TaoLv) [inline scripts/bert/finetune_classifier.py]: move this to for loop? we need measure the time of each epoch like what did in train function.

File: scripts/bert/index.rst [hunk 3]
```diff
@@ -45,4 +45,12 @@ It gets validation accuracy of `88.7% <https://raw.githubusercontent.com/dmlc/we
 It gets RTE validation accuracy of `70.8% <https://raw.githubusercontent.com/dmlc/web-data/master/gluonnlp/logs/bert/finetuned_rte.log>`_
 , whereas the the original Tensorflow implementation give evaluation results 66.4%.
 
-Some other tasks can be modeled with `--task_name` parameter.
\ No newline at end of file
+Use the following command to make inference of fine-tuned BERT model for classification on the GLUE(MRPC) dataset.
+
+.. code-block:: console
+
+   $ MXNET_GPU_MEM_POOL_TYPE=Round GLUE_DIR=glue_data python3 finetune_classifier.py --task_name MRPC --batch_size 32 --run_type inference
```
Comment (TaoLv) [inline scripts/bert/index.rst]: remove `MXNET_GPU_MEM_POOL_TYPE=Round` if `--gpu` is not present.

**Comment by TaoLv** [comment]
I changed the default value of pad to true in this PR. Hope you don't mind. @pengxin99

File: scripts/bert/finetune_classifier.py (hunk 1 above)
Comment (pengxin99) [reply scripts/bert/finetune_classifier.py:355]: > 
> 
> why remove this?

at the start of epoch, tic will be reassignment by tic=time.time(), so i think here is invalid, and remove this.

**Comment by pengxin99** [comment]
> 
> 
> I changed the default value of pad to true in this PR. Hope you don't mind. @pengxin99

it is ok :)

**Review by eric-haibin-lin — APPROVED** [review]
Looks good. We should also evaluate larger max_len such as 384, and smaller batch size such as 1

**Comment by TaoLv** [comment]
Sure. So the basic batch size set is [1, 128, 256, 384, 512]. Will use 1 for inference evaluation and others for training. Does this make sense to you, @eric-haibin-lin ?

**Comment by TaoLv** [comment]
For inference,  bs=1 for latency and bs=2^N for throughput measurement.
